torch/_python_dispatcher.py

```diff
@@ -0,0 +1,154 @@
+import re
+import torch._C as C
+
+
+"""
+PythonDispatcher class is a thin python-binding to C++ dispatcher and it
+is designed to show how dispatcher precompute works. In particular,
+it shows for a certain op `foo`, how the computed dispatch table looks
+like after user register their kernels to certains dispatch keys.
+
+In the real C++ dispatcher we support many dispatch keys for different
+functionalities. For simplicity PythonDispatcher only support one example
+dispatch key in each category.
```

Comment:
I'm not sure how to relate the "one example dispatch key" here to the pairs of dispatch keys below. Maybe if you said "For simplicity PythonDispatcher only supports dispatch keys for a single example of each use case. These use cases are listed below." Something like that?